arduino-ide-extension/src/test/node/core-service-impl.slow-test.ts

```diff
@@ -0,0 +1,316 @@
+          ([command]) =>
+            command === 'arduino.languageserver.notifyBuildDidComplete'
+        );
+      expect(executedBuildDidCompleteCommands.length).to.be.equal(1);
+      const [, args] = executedBuildDidCompleteCommands[0];
+      expect(args.length).to.be.equal(1);
+      const arg = args[0];
+      expect(typeof arg).to.be.equal('object');
+      expect('buildOutputUri' in arg).to.be.true;
+      expect(arg.buildOutputUri).to.be.not.undefined;
+
+      const tempBuildPaths = await sketchesService.tempBuildPath(sketch);
+      if (isWindows) {
+        expect(tempBuildPaths.length).to.be.greaterThan(1);
+      } else {
+        expect(tempBuildPaths.length).to.be.equal(1);
+      }
+
+      const { buildOutputUri } = arg;
+      const buildOutputPath = FileUri.fsPath(buildOutputUri).toString();
+      expect(tempBuildPaths.includes(buildOutputPath)).to.be.true;
+    });
+  });
+});
+
+async function start(
+  container: Container,
+  toDispose: Disposable[]
+): Promise<void> {
+  const daemon = container.get<ArduinoDaemonImpl>(ArduinoDaemonImpl);
+  const configService = container.get<ConfigServiceImpl>(ConfigServiceImpl);
+  toDispose.push(Disposable.create(() => daemon.stop()));
+  configService.onStart();
+  daemon.onStart();
+  await waitForEvent(daemon.onDaemonStarted, 10_000);
+  const boardService = container.get<BoardsServiceImpl>(BoardsServiceImpl);
+  const searchResults = await boardService.search({ query: avr });
+  const platform = searchResults.find(({ id }) => id === avr);
+  if (!platform) {
+    throw new Error(`Could not find platform: ${avr}`);
+  }
+  await boardService.install({ item: platform });
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars, unused-imports/no-unused-vars
+function createContainer(): Container {
+  const container = new Container({ defaultScope: 'Singleton' });
+  const module = new ContainerModule((bind) => {
+    bind(CoreClientProvider).toSelf().inSingletonScope();
+    bind(CoreServiceImpl).toSelf().inSingletonScope();
+    bind(CoreService).toService(CoreServiceImpl);
+    bind(BoardsServiceImpl).toSelf().inSingletonScope();
+    bind(BoardsService).toService(BoardsServiceImpl);
+    bind(TestResponseService).toSelf().inSingletonScope();
+    bind(ResponseService).toService(TestResponseService);
+    bind(MonitorManager).toSelf().inSingletonScope();
+    bind(MonitorServiceFactory).toFactory(
+      ({ container }) =>
+        (options: MonitorServiceFactoryOptions) => {
+          const child = container.createChild();
+          child
+            .bind<MonitorServiceFactoryOptions>(MonitorServiceFactoryOptions)
+            .toConstantValue({
+              ...options,
+            });
+          child.bind(MonitorService).toSelf();
+          return child.get<MonitorService>(MonitorService);
+        }
+    );
+    bind(EnvVariablesServer).toSelf().inSingletonScope();
+    bind(TheiaEnvVariablesServer).toService(EnvVariablesServer);
```

Comment:
It looks incorrect:

```
ino-ide-extension: Configuration directory URI: 'file:///c%3A/Users/kittaakos/.theia'
```

Reply:
It's some trash logged to the console. It's from [here](https://github.com/eclipse-theia/theia/blob/3b4fb797eb9908f9c0b9f2e3fafa0255ee8054fe/packages/core/src/node/env-variables/env-variables-server.ts#L34). The tests eventually do not use this location. Check the existence of `~/.theia/arduino-cli.yaml`; it's not there. Run the tests: `yarn test:slow`, and check again `~/.theia/arduino-cli.yaml`. It's still not there.